README.md
# website-template

## Build Setup

```bash
# install dependencies
$ npm install

# serve with hot reload at localhost:3000
$ npm run dev

# build for production and launch server
$ npm run build
$ npm run start

# generate static project
$ npm run generate
```

For detailed explanation on how things work, check out the [documentation](https://nuxtjs.org).

## Special Directories

You can create the following extra directories, some of which have special behaviors. Only `pages` is required; you can delete them if you don't want to use their functionality.

### `assets`

The assets directory contains your uncompiled assets such as Stylus or Sass files, images, or fonts.

More information about the usage of this directory in [the documentation](https://nuxtjs.org/docs/2.x/directory-structure/assets).

### `components`

The components directory contains your Vue.js components. Components make up the different parts of your page and can be reused and imported into your pages, layouts and even other components.

More information about the usage of this directory in [the documentation](https://nuxtjs.org/docs/2.x/directory-structure/components).

### `layouts`

Layouts are a great help when you want to change the look and feel of your Nuxt app, whether you want to include a sidebar or have distinct layouts for mobile and desktop.

More information about the usage of this directory in [the documentation](https://nuxtjs.org/docs/2.x/directory-structure/layouts).


### `pages`

This directory contains your application views and routes. Nuxt will read all the `*.vue` files inside this directory and setup Vue Router automatically.

More information about the usage of this directory in [the documentation](https://nuxtjs.org/docs/2.x/get-started/routing).

### `plugins`

The plugins directory contains JavaScript plugins that you want to run before instantiating the root Vue.js Application. This is the place to add Vue plugins and to inject functions or constants. Every time you need to use `Vue.use()`, you should create a file in `plugins/` and add its path to plugins in `nuxt.config.js`.

More information about the usage of this directory in [the documentation](https://nuxtjs.org/docs/2.x/directory-structure/plugins).

### `static`

This directory contains your static files. Each file inside this directory is mapped to `/`.

Example: `/static/robots.txt` is mapped as `/robots.txt`.

More information about the usage of this directory in [the documentation](https://nuxtjs.org/docs/2.x/directory-structure/static).

### `store`

This directory contains your Vuex store files. Creating a file in this directory automatically activates Vuex.

More information about the usage of this directory in [the documentation](https://nuxtjs.org/docs/2.x/directory-structure/store).


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Unreleased]

[unreleased]: https://gitlab.com/the-seed-squad/templates/website/compare/master...develop


nuxt.config.js
export default {
  target: "static",
  publicRuntimeConfig: {
    baseUrl: process.env.DOMAIN ? `https://${process.env.DOMAIN}` : "http://localhost:3000",
    assetsUrl: process.env.DOMAIN ? `https://api.${process.env.DOMAIN}/assets` : "http://localhost:8080/assets",
    n8nUrl: process.env.DOMAIN ? `https://n8n.${process.env.DOMAIN}` : "http://localhost:5678"
  },
  head: {
    title: "Template",
    htmlAttrs: {
      lang: "es"
    },
    meta: [
      { charset: "utf-8" },
      { name: "viewport", content: "width=device-width, initial-scale=1" },
      { hid: "description", name: "description", content: "" },
      { name: "format-detection", content: "telephone=no" }
    ],
    link: [{ rel: "icon", type: "image/x-icon", href: "/favicon.ico" }]
  },
  build: {},
  buildModules: ["@nuxtjs/tailwindcss", "@nuxtjs/google-fonts"],
  modules: ["@nuxtjs/axios"],
  plugins: ["~/plugins/webhook"],
  content: {},
  components: true,
  googleFonts: {
    display: "swap",
    families: {
      Roboto: [100, 200, 300, 400, 500, 600, 700, 800, 900]
    }
  },
  tailwindcss: {
    cssPath: "tailwind.css",
    configPath: "tailwind.config.js",
    exposeConfig: false
  },
  axios: {
    baseURL: process.env.DOMAIN ? `https://api.${process.env.DOMAIN}` : "http://localhost:8080"
  }
};


store/modals.js
export const state = () => ({
  index: 0,
  items: []
});

export const getters = {};

export const mutations = {
  RESET(state) {
    state.index = 0;
    state.items = [];
  },
  ADD(state, modal) {
    modal.id = state.index += 1;
    modal.active = true;
    state.items.push(modal);
  },
  REMOVE(state, modal) {
    const index = state.items.findIndex(({ id }) => modal.id === id);
    state.items.splice(index, 1);
  },
  ACTIVATE(state, modal) {
    modal.active = true;
  },
  DEACTIVATE(state, modal) {
    modal.active = false;
  }
};

export const actions = {
  reset({ commit }) {
    commit("RESET");
  },
  open({ commit }, { component, props, events }) {
    const modal = { component, props, events };
    commit("ADD", modal);
  },
  close({ commit }, modal) {
    commit("DEACTIVATE", modal);

    setTimeout(() => {
      commit("REMOVE", modal);
    }, 300);
  },
  activate({ commit }, modal) {
    commit("ACTIVATE", modal);
  },
  deactivate({ commit }, modal) {
    commit("DEACTIVATE", modal);
  }
};

export default {
  namespaced: true,
  state,
  getters,
  actions,
  mutations
};


tailwind.config.js
const defaultTheme = require("tailwindcss/defaultTheme");
const plugin = require("tailwindcss/plugin");

const colors = {
  yellow: {
    700: "#FFCE00",
    500: "#FFDC48",
    300: "#FFE783",
    100: "#FFF1B7"
  },
  pink: {
    700: "#DE2085",
    500: "#E74099",
    300: "#F173B6",
    100: "#FFB5DC"
  },
  green: {
    700: "#71B739",
    500: "#8BCE55",
    300: "#A2E170",
    100: "#C1EF9C"
  },
  black: {
    900: "#1F1F1F",
    700: "#333333",
    500: "#666666",
    300: "#B3B3B3",
    100: "#DAD8D8"
  }
};

module.exports = {
  mode: "jit",
  darkMode: false,
  theme: {
    extend: {
      fontFamily: {
        body: ["Roboto", ...defaultTheme.fontFamily.sans]
      },
      backgroundColor: {
        body: "#F9F9F9",
        ...colors
      },
      colors: {
        ...colors
      },
      boxShadow: {
        down: "-3px 3px 0px rgba( 0, 0, 0, 1)",
        top: "-4px -4px 0px rgba( 0, 0, 0, 1)"
      },
      fontSize: {
        title: "40px"
      },
      letterSpacing: {
        title: "-1.5px",
        button: "-1px",
        text: "-0.25px"
      },
      scale: {
        "-1": "-1"
      },
      rotate: {
        "-21": "-21.7deg"
      }
    }
  },
  variants: {
    extend: {}
  },
  plugins: [
    require("@tailwindcss/aspect-ratio"),
    require("@tailwindcss/forms"),
    require("@tailwindcss/line-clamp"),
    require("@tailwindcss/typography"),
    plugin(({ addBase, theme }) => {
      addBase({
        body: {
          backgroundColor: theme("backgroundColor.body"),
          fontFamily: theme("fontFamily.body")
        }
      });
    })
  ],
  purge: {
    enabled: process.env.NODE_ENV === "production",
    content: ["components/**/*.vue", "layouts/**/*.vue", "pages/**/*.vue", "plugins/**/*.js", "nuxt.config.js"]
  }
};
